src/utils/staking.js

```diff
@@ -275,23 +277,39 @@ export class Staking {
     }
 
     async getValidators(accountIds) {
+        const { current_validators, next_validators, current_proposals } = await this.provider.validators()
+        const currentValidators = current_validators.map(({ account_id }) => account_id)
+        const nextValidators = current_proposals.map(({ account_id }) => account_id)
+        
         if (!accountIds) {
-            const { current_validators, next_validators, current_proposals } = await this.provider.validators()
-            const allValidators = [...current_validators, ...next_validators, ...current_proposals]
-            accountIds = [...new Set(allValidators.map(({ account_id }) => account_id))]
+            const rpcValidators = [...current_validators, ...next_validators, ...current_proposals].map(({ account_id }) => account_id)
+
+            // TODO use indexer - getting all historic validators from raw GH script .json
+            const networkId = this.provider.connection.url.indexOf('mainnet') > -1 ? 'mainnet' : 'testnet'
```

Comment:
`this.provider.connection.networkId`

Reply:
showing as undefined...?

Reply:
ah, looks like `this.provider.connection` is a scam, not an actual `Connection` instance. I guess given it's in wallet, reasonable to use `NETWORK_ID` as well.